### ISSUE TITLE
Helper function for automateing model fitting

### ISSUE THREAD

**Issue by Balandat** [cornellius-gp/gpytorch#279]
Currently, model fitting requires writing a custom loop to take gradient steps. All the tutorials perform that for a fixed number of steps without any kind of smartness about how the loss / the parameter values evolve during the course of the optimization.

To make this easier for the non-expert (and to avoid writing tons of boilerplate code), I suggest we provide an (optional) wrapper for the model fitting loop that uses some reasonable convergence criteria to stop the optimization.

Given a specified `model`, `likelihood`, `mll_class`, `train_x` and `train_y`, calling this could be as simple as:
```
from gpytorch import fit_model
fit_model(model, likelihood, train_x, train_y, mll_class)
```

Implementation wise this would be something like the following:
```
def fit_model(model, likelihood, train_x, train_y, mll_class, optimizer=None, options=None, verbose=True):
    options = options or {}
    model.train()
    likelihood.train()
    mll = mll_class(likelihood, model)
    optimizer = optimizer or torch.optim.Adam(
        [{"params": model.parameters()}],
        lr=options.get("lr", 0.1), # be smarter about this
    )
    param_trajectory = {
        name: [] for name, param in model.named_parameters()
    }
    loss_trajectory = []
    converged = False
    while not converged:
        optimizer.zero_grad()
        output = model(train_x)
        loss = -mll(output, train_y)
        losses.append(loss.data.clone())
        loss.backward()
        if verbose:
            print("Iter {} - MLL: {:.3f}".format(i + 1, -loss.item()))
        optimizer.step()
        for name, param in model.named_parameters():
            param_trajectory[name].append(param.data.clone())
        converged = check_convergence(loss_trajectory, param_trajectory)
    model.eval()
    likelihood.eval()
```

Of course we'd need to think a little harder about what `check_convergence` should be for the stochastic optimization algorithms used here.

**Comment by jacobrgardner** [comment]
An intern here was working over the summer on a scikit learn / skorch style wrapper for GPyTorch. I believe this is actually in a pretty good state, and might also be something we will throw in a public repo fairly soon.

**Comment by Balandat** [comment]
Ah ok - would be good to see what's there. Would the idea be to bundle this with gpytorch or have it separate?
